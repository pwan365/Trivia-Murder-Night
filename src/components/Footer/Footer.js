import { Grid, Icon, Typography } from '@mui/material';
import React from 'react';

import InstagramIcon from '@mui/icons-material/Instagram';
import FacebookIcon from '@mui/icons-material/Facebook';

const Footer = () => {
  return (
    <Grid style={{height:'200px', width:'100%', background: 'black', display:'flex', flexDirection:'column', justifyContent:'center', alignItems:'center'}}>
      <Grid>
        <Typography color={"white"} style={{marginBottom:'16px'}}>
          Follow Us For Daily Updates!
        </Typography>
      </Grid>
      <Grid style={{display:'flex', justifyContent:'center', alignItems:'center'}}>
        <InstagramIcon fontSize="large" sx={{ color: 'white', marginRight: '30px'}}/>
        <FacebookIcon fontSize="large" sx={{ color: 'white' }}/>
      </Grid>
    </Grid>
  )

}



export default Footer;