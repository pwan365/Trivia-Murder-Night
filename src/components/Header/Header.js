import React, { useState } from "react";
import {
  AppBar,
  Box,
  Button,
  Divider,
  Drawer,
  Grid,
  Hidden,
  IconButton,
  List,
  ListItem,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import DuckIcon from "../../resources/duck.png";
import { ArrowForward } from "@mui/icons-material";
import { Link } from "react-router-dom";

let headerButtons = [
  {
    title: "Home",
    path: "/",
  },
  {
    title: "Tutorial",
    path: "/tutorial",
  },
];

const Header = () => {
  const [toggleSideMenu, setToggleSideMenu] = useState(false);

  const toggleMenu = () => {
    setToggleSideMenu(!toggleSideMenu);
  };

  return (
    <div>
      <AppBar
        style={{ background: "transparent", color: "white" }}
        elevation={0}
      >
        <Toolbar
          style={{
            display: "flex",
            justifyContent: "space-between",
          }}
        >
          <IconButton>
            <img style={{ width: "40px" }} src={DuckIcon} />
          </IconButton>

          <Hidden mdDown>
            <Grid>
              {headerButtons.map((item) => {
                return (
                  <Link to={item.path} style={{ textDecoration: "none" }}>
                    <Button key={item.title} style={{ marginRight: "20px" }}>
                      <Typography style={{ color: "white" }}>
                        {item.title}
                      </Typography>
                    </Button>
                  </Link>
                );
              })}
            </Grid>
          </Hidden>

          <Hidden mdUp>
            <IconButton onClick={toggleMenu} color="inherit">
              <MenuIcon />
            </IconButton>
          </Hidden>
        </Toolbar>
      </AppBar>

      <Drawer
        anchor={"right"}
        open={toggleSideMenu}
        onClose={toggleMenu}
        variant="persistent"
      >
        <Box
          role="presentation"
          style={{ width: "140px" }}
          onClick={toggleMenu}
        >
          <List>
            <ListItem button>
              <ArrowForward />
            </ListItem>
            <Divider />
            {headerButtons.map((item) => (
              <Link
                to={item.path}
                style={{ textDecoration: "none", color: "black" }}
              >
                <ListItem button key={item.title}>
                  <ListItemText primary={item.title} />
                </ListItem>
              </Link>
            ))}
          </List>
        </Box>
      </Drawer>
    </div>
  );
};
export default Header;

<Toolbar></Toolbar>;
