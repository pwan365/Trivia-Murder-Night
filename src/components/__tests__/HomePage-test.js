import {MemoryRouter} from 'react-router'
import React from 'react'
import {create} from 'react-test-renderer'
import Footer from '../Footer/Footer'
import Header from '../Header/Header'
import * as renderer from 'react-test-renderer'

const footer = create (<Footer />);


test('snapshot', () => {
    expect(footer).toMatchSnapshot();
});

test('snapshot', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});