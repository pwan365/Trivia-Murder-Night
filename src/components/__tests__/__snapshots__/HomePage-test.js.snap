// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="MuiGrid-root css-vj1n65-MuiGrid-root"
  style={
    Object {
      "alignItems": "center",
      "background": "black",
      "display": "flex",
      "flexDirection": "column",
      "height": "200px",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
      style={
        Object {
          "marginBottom": "16px",
        }
      }
    >
      Follow Us For Daily Updates!
    </p>
  </div>
  <div
    className="MuiGrid-root css-vj1n65-MuiGrid-root"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1m5jkh2-MuiSvgIcon-root"
      data-testid="InstagramIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-f074gz-MuiSvgIcon-root"
      data-testid="FacebookIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
      />
    </svg>
  </div>
</div>
`;

exports[`snapshot 2`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1hc7nu0-MuiPaper-root-MuiAppBar-root"
    style={
      Object {
        "background": "transparent",
        "color": "white",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          src="duck.png"
          style={
            Object {
              "width": "40px",
            }
          }
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        className="MuiBox-root css-0"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "width": "140px",
          }
        }
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <div
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <hr
            className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Home
                </span>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            href="/tutorial"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Tutorial
                </span>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </a>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
